x-pack/plugins/siem/public/store/timeline/reducer.test.ts

```diff
@@ -1706,5 +1710,58 @@ describe('Timeline', () => {
       };
       expect(update).toEqual(expected);
     });
+
+    test('should remove only first provider and not nested andProvider', () => {
```

Comment:
Thanks for this test! Consider this additional test for the case where there are multiple ANDed providers:

```
  test('should remove only the first provider and keep multiple nested andProviders', () => {
    const multiDataProvider: DataProvider[] = [
      {
        and: [
          {
            enabled: true,
            id: 'socket_closed-MSoH7GoB9v5HJNSHRYj1-user_name-root',
            name: 'root',
            excluded: false,
            kqlQuery: '',
            queryMatch: {
              field: 'user.name',
              value: 'root',
              operator: ':',
            },
          },
          {
            enabled: true,
            id: 'executed-yioH7GoB9v5HJNSHKnp5-auditd_result-success',
            name: 'success',
            excluded: false,
            kqlQuery: '',
            queryMatch: {
              field: 'auditd.result',
              value: 'success',
              operator: ':',
            },
          },
        ],
        enabled: true,
        excluded: false,
        id: 'hosts-table-hostName-suricata-iowa',
        name: 'suricata-iowa',
        kqlQuery: '',
        queryMatch: {
          field: 'host.name',
          value: 'suricata-iowa',
          operator: ':',
        },
      },
    ];

    const multiDataProviderMock = set('foo.dataProviders', multiDataProvider, timelineByIdMock);

    const update = removeTimelineProvider({
      id: 'foo',
      providerId: 'hosts-table-hostName-suricata-iowa',
      timelineById: multiDataProviderMock,
    });

    expect(update).toEqual(
      set(
        'foo.dataProviders',
        [
          {
            enabled: true,
            id: 'socket_closed-MSoH7GoB9v5HJNSHRYj1-user_name-root',
            name: 'root',
            excluded: false,
            kqlQuery: '',
            queryMatch: {
              field: 'user.name',
              value: 'root',
              operator: ':',
            },
            and: [
              {
                enabled: true,
                id: 'executed-yioH7GoB9v5HJNSHKnp5-auditd_result-success',
                name: 'success',
                excluded: false,
                kqlQuery: '',
                queryMatch: {
                  field: 'auditd.result',
                  value: 'success',
                  operator: ':',
                },
              },
            ],
          },
        ],
        timelineByIdMock
      )
    );
  });
```

Reply:
Added! 👍

Reply:
The two additional (suggested) unit tests below would add coverage for `removeTimelineProvider`'s branching logic into `removeAndProvider`: 

```
  test('should remove only the first AND provider when the first AND is deleted, and there are multiple andProviders', () => {
    const multiDataProvider: DataProvider[] = [
      {
        and: [
          {
            enabled: true,
            id: 'socket_closed-MSoH7GoB9v5HJNSHRYj1-user_name-root',
            name: 'root',
            excluded: false,
            kqlQuery: '',
            queryMatch: {
              field: 'user.name',
              value: 'root',
              operator: ':',
            },
          },
          {
            enabled: true,
            id: 'executed-yioH7GoB9v5HJNSHKnp5-auditd_result-success',
            name: 'success',
            excluded: false,
            kqlQuery: '',
            queryMatch: {
              field: 'auditd.result',
              value: 'success',
              operator: ':',
            },
          },
        ],
        enabled: true,
        excluded: false,
        id: 'hosts-table-hostName-suricata-iowa',
        name: 'suricata-iowa',
        kqlQuery: '',
        queryMatch: {
          field: 'host.name',
          value: 'suricata-iowa',
          operator: ':',
        },
      },
    ];

    const multiDataProviderMock = set('foo.dataProviders', multiDataProvider, timelineByIdMock);

    const update = removeTimelineProvider({
      andProviderId: 'socket_closed-MSoH7GoB9v5HJNSHRYj1-user_name-root',
      id: 'foo',
      providerId: 'hosts-table-hostName-suricata-iowa',
      timelineById: multiDataProviderMock,
    });

    expect(update).toEqual(
      set(
        'foo.dataProviders',
        [
          {
            and: [
              {
                enabled: true,
                id: 'executed-yioH7GoB9v5HJNSHKnp5-auditd_result-success',
                name: 'success',
                excluded: false,
                kqlQuery: '',
                queryMatch: {
                  field: 'auditd.result',
                  value: 'success',
                  operator: ':',
                },
              },
            ],
            enabled: true,
            excluded: false,
            id: 'hosts-table-hostName-suricata-iowa',
            name: 'suricata-iowa',
            kqlQuery: '',
            queryMatch: {
              field: 'host.name',
              value: 'suricata-iowa',
              operator: ':',
            },
          },
        ],
        timelineByIdMock
      )
    );
  });

  test('should remove only the second AND provider when the second AND is deleted, and there are multiple andProviders', () => {
    const multiDataProvider: DataProvider[] = [
      {
        and: [
          {
            enabled: true,
            id: 'socket_closed-MSoH7GoB9v5HJNSHRYj1-user_name-root',
            name: 'root',
            excluded: false,
            kqlQuery: '',
            queryMatch: {
              field: 'user.name',
              value: 'root',
              operator: ':',
            },
          },
          {
            enabled: true,
            id: 'executed-yioH7GoB9v5HJNSHKnp5-auditd_result-success',
            name: 'success',
            excluded: false,
            kqlQuery: '',
            queryMatch: {
              field: 'auditd.result',
              value: 'success',
              operator: ':',
            },
          },
        ],
        enabled: true,
        excluded: false,
        id: 'hosts-table-hostName-suricata-iowa',
        name: 'suricata-iowa',
        kqlQuery: '',
        queryMatch: {
          field: 'host.name',
          value: 'suricata-iowa',
          operator: ':',
        },
      },
    ];

    const multiDataProviderMock = set('foo.dataProviders', multiDataProvider, timelineByIdMock);

    const update = removeTimelineProvider({
      andProviderId: 'executed-yioH7GoB9v5HJNSHKnp5-auditd_result-success',
      id: 'foo',
      providerId: 'hosts-table-hostName-suricata-iowa',
      timelineById: multiDataProviderMock,
    });

    expect(update).toEqual(
      set(
        'foo.dataProviders',
        [
          {
            and: [
              {
                enabled: true,
                id: 'socket_closed-MSoH7GoB9v5HJNSHRYj1-user_name-root',
                name: 'root',
                excluded: false,
                kqlQuery: '',
                queryMatch: {
                  field: 'user.name',
                  value: 'root',
                  operator: ':',
                },
              },
            ],
            enabled: true,
            excluded: false,
            id: 'hosts-table-hostName-suricata-iowa',
            name: 'suricata-iowa',
            kqlQuery: '',
            queryMatch: {
              field: 'host.name',
              value: 'suricata-iowa',
              operator: ':',
            },
          },
        ],
        timelineByIdMock
      )
    );
  });
```

Reply:
Giddy up -- thanks for the additional coverage! I went ahead and added these along with some additional testing for a missed scenario where previous root filters were not being checked.